pkg/kubernetes/kube.go

```diff
@@ -722,6 +747,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
+			}
+		}
+		return nil
+	}, time.Duration(timeout)*time.Second).Should(gomega.Succeed(), func() string {
+		return fmt.Sprintf("Pod has no %s message in the logs", searchkeyword)
+	})
+	return nil
+}
+
+// ListPodsWithLabelSelector lists pods with a label selector
+func listPodsWithLabelSelector(client kubernetes.Interface, namespace, selector string) (*corev1.PodList, error) {
+	pods, err := retryOnError(&DefaultRetry, isRetriable, func() (interface{}, error) {
+		return client.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: selector})
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list pods")
+	}
+
+	return pods.(*corev1.PodList), nil
+}
+
+// TODO: Move this when other prs are merged in
+type condFunc func() (interface{}, error)
+
+// TODO: Move this when other prs are merged in
+func retryOnError(backoff *wait.Backoff, retryExpected func(error) bool, fn condFunc) (interface{}, error) {
+	var ex, lastErr error
+	var out interface{}
+	caller := runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name()
+	err := wait.ExponentialBackoff(*backoff, func() (bool, error) {
+		out, ex = fn()
+		switch {
+		case ex == nil:
+			return true, nil
+		case retryExpected(ex):
+			lastErr = ex
+			log.Warnf("A caller %v retried due to exception: %v", caller, ex)
+			return false, nil
+		default:
+			return false, ex
+		}
+	})
+	if err == wait.ErrWaitTimeout {
+		err = lastErr
+	}
+	return out, err
+}
+
+// TODO: Move this when other prs are merged in
+func isRetriable(err error) bool {
+	for _, msg := range retriableErrors {
+		if strings.Contains(err.Error(), msg) {
+			return true
+		}
+	}
+	return false
+}
+
+func findStringInPodLogs(kc *Client, pod corev1.Pod,
+	since time.Time, stringsToFind ...string) (int, error) {
+
+	var sinceTime metav1.Time = metav1.NewTime(since)
+
+	foundCount := 0
+
+	for _, container := range pod.Spec.Containers {
+		podLogOpts := corev1.PodLogOptions{
+			SinceTime: &sinceTime,
+			Container: container.Name,
+		}
+
+		req := kc.KubeInterface.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+		podLogs, err := req.Stream(context.Background())
+		if err != nil {
+			return 0, errors.Errorf("Error in opening stream for pod %s, container %s : %s", pod.Name, container.Name, string(err.Error()))
+		}
+
+		scanner := bufio.NewScanner(podLogs)
+		for scanner.Scan() {
+			line := scanner.Text()
+			for _, stringToFind := range stringsToFind {
+				if strings.Contains(line, stringToFind) {
+					foundCount += 1
+					log.Infof("Found matching string in line: '%s'", line)
+				}
+			}
+		}
+		_ = podLogs.Close()
+	}
+
+	return foundCount, nil
+}
+
+func (kc *Client) NoMatchingStringInLogsSinceTime(namespace,
+	selector, searchkeyword, sinceTime string) error {
+
+	if kc.KubeInterface == nil {
+		return errors.Errorf("'Client.KubeInterface' is nil. 'AKubernetesCluster' sets this interface, try calling it before using this method")
+	}
```

Comment:
Nothing wrong with this currently, but Alfredo's pending PR has a kc.Validate() function that should replace this